README.md

```diff
@@ -470,6 +470,63 @@ credentials unless they do not meet your specific requirements.
 You can now [use the Auth library](#using-external-identities) to call Google Cloud
 resources from an OIDC or SAML provider.
 
+#### Using a custom supplier with OIDC and SAML
+A custom implementation of IdentityPoolSubjectTokenSupplier can be used while building IdentityPoolCredentials
+to supply a subject token which can be exchanged for a GCP access token.
+
+```java
+class TokenSupplier implements IdentityPoolSubjectTokenSupplier {
+  @Override
+  String getSubjectToken(){
+    // return a valid subject token for the configured identity.
```

Comment:
Can you include some plausible code to fetch a token here and below, including error handling / retry case?

Reply:
Can add some usable samples, but I think it would be preferable to have them in the samples folder vs the entire implemented class in the readme. Especially since it would probably be valuable to have a few sample options. Also don't think that should block the basic documentation, vs something we add in the near future.

Reply:
It can be pretty basic; like:

```
@Override
public String getSubjectToken() throws IOException {
  try {
    // Get token.
  } catch (Exception e) {
    // throw IOException on errors, maybe explain retry policy if any.
  }
}
```

I suspect lots of people will copy and paste examples from README.md before they go dig into samples, so it's worth at least giving the structure that will make them successful.

Reply:
Added more detailed example